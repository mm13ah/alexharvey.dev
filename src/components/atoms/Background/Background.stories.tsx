import React from 'react';
import { StoryFn, Meta } from '@storybook/react';

import { Background } from './Background';

export default {
  title: 'Atoms/Background',
  component: Background,
  parameters: {
    layout: 'fullscreen',
  },
  args: {
    className: 'bg-gradient-to-r from-cyan-500 to-blue-500',
  },
} as Meta<typeof Background>;

const Template: StoryFn<typeof Background> = (args) => <Background {...args} />;

export const Primary = Template.bind({});
Primary.storyName = 'Background';
