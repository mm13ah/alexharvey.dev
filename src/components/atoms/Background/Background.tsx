import React from 'react';

interface Props {
  id?: string;
  className?: string;
  children?: React.ReactNode;
}

export const Background = ({ id, className, children }: Props) => (
  <section id={id} className={`w-screen min-h-screen flex ${className}`}>
    {children}
  </section>
);
