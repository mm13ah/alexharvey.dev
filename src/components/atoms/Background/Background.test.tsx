import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Background } from './Background';

describe('Background', () => {
  it('should render', () => {
    render(<Background />);
  });
  it('should match a snapshot', () => {
    const tree = renderer.create(<Background />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
