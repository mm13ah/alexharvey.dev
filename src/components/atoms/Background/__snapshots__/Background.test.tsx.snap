// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Background should match a snapshot 1`] = `
<section
  className="w-screen min-h-screen flex undefined"
/>
`;
